### DISCUSSION TITLE
pynetbox: Get device data inclunding interface data

### DISCUSSION THREAD

**Discussion by misch42** [netbox-community/netbox#12266] [Help Wanted!] [answered]
Hi,

how do I retrieve device data with pynetbox including the interface information of the device?

I have:
device = nb.dcim.devices.get(serial='XXXX')

and I get the usual API view output. But there a possibility to retreive the interface data also? 

Michael.

**Comment by markkuleinio** [comment, marked as answer]
Example:

```
>>> device = nb.dcim.devices.get(name="Test")
>>> device
Test
>>> interfaces = list(nb.dcim.interfaces.filter(device_id=device.id))
>>> interfaces
[Loopback, eth0]
>>>
```

**Reply by misch42** [reply]
Thanks. That is a good solution. Works for me.
